language/tools/move-unit-test/src/test_reporter.rs

```diff
@@ -219,6 +226,87 @@ impl TestFailure {
+            .get(&loc.file_hash())
+            .ok_or(codespan_reporting::files::Error::FileMissing)?;
+        let start_line_index = files.line_index(*id, loc.start() as usize)?;
+        let start_line_number = files.line_number(*id, start_line_index)?;
+        let end_line_index = files.line_index(*id, loc.end() as usize)?;
+        let end_line_number = files.line_number(*id, end_line_index)?;
+        if start_line_number == end_line_number {
+            Ok(start_line_number.to_string())
+        } else {
+            Ok(format!("{}-{}", start_line_number, end_line_number))
+        }
+    }
+
+    fn report_exec_state(test_plan: &TestPlan, exec_state: &ExecutionState) -> String {
+        let stack_trace = exec_state.stack_trace();
+        let mut buf = String::new();
+        if !stack_trace.is_empty() {
+            buf.push_str("stack trace\n");
+            let mut files = SimpleFiles::new();
+            let mut file_mapping = HashMap::new();
+            for (fhash, (fname, source)) in &test_plan.files {
+                let id = files.add(*fname, source.as_str());
+                file_mapping.insert(*fhash, id);
+            }
+
+            for frame in stack_trace {
+                if let Some(module_id) = &frame.0 {
+                    if let Some(named_module) = test_plan.module_info.get(module_id) {
+                        if let Ok(function_source_map) =
+                            named_module.source_map.get_function_source_map(frame.1)
+                        {
+                            // unwrap here is a mirror of the same unwrap in report_error_with_location
+                            let loc = function_source_map.get_code_location(frame.2).unwrap();
+                            let fn_handle_idx =
+                                named_module.module.function_def_at(frame.1).function;
+                            let fn_id_idx =
+                                named_module.module.function_handle_at(fn_handle_idx).name;
+                            let fn_name = named_module.module.identifier_at(fn_id_idx).as_str();
+                            let file_name = match test_plan.files.get(&loc.file_hash()) {
+                                Some(v) => format!("{}", v.0),
+                                None => "unknown_source".to_string(),
+                            };
+                            buf.push_str(
+                                &format!(
+                                    "\t{}::{}({}:{})\n",
+                                    module_id.name(),
+                                    fn_name,
+                                    file_name,
+                                    Self::get_line_number(&loc, &files, &file_mapping)
+                                )
+                                .to_string(),
+                            );
+                        } else {
+                            return "\tmalformed stack trace (no source map)".to_string();
```

Comment:
Actually, since you are returning in all of these you should do 
```
let function_souce_map = match named_module.source_map.get_function_source_map(frame.1) {
  Some(function_souce_map) => function_souce_map,
  None => return "\tmalformed stack trace (no source map)".to_string();
};
```